.github/workflows/ci.yml

```
@@ -2,127 +2,454 @@ name: CI
+            --output trivy-results.sarif .
+
+          # Run Gitleaks with configuration
+          gitleaks detect \
+            --verbose \
+            --report-format sarif \
+            --report-path gitleaks-results.sarif \
+            --config .gitleaks.toml
+
+      - name: Upload Trivy results
+        if: always()
+        uses: actions/upload-artifact@v3
```

actionlint (1.7.4)

the runner of "actions/upload-artifact@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)